### ISSUE TITLE
Home/User Folder in the systems directory

### ISSUE THREAD

**Issue by Papermanzero** [libretro/citra#26]
currently the necessary system files are located in the user directoy of the OS.
However citra shoulf behave like ppsspp which uses the systems folder within the retroarch directory for bios files/user files
Means

- RetroArch
           - System
                   - PPSSPP

**Comment by shinra358** [comment]
+1

**Comment by i30817** [comment]
I remember changing the path on the retroarch.cfg on linux of rename several path variables after i moved the config dir with "XDG_CONFIG_HOME" so there are other candidates on the same boat.

from diffiing my own current cfg and a new one i see several candidates that might need the same idea on the new cfg, old file entry first, new file entry second ( <retroarch> is not a real pathname, it's just my xdg override with the name 'retroarch representing it will create the dir there):


rgui_config_directory = "<retroarch>/config/"

rgui_config_directory = "~/.config/retroarch/config/"



content_database_path = "<retroarch>/database/rdb"

content_database_path = "<//retroarch>database/rdb"  Uh? It seems a additional / sneaked in on this setting.

cheat_database_path = "<retroarch>/cheats"

cheat_database_path = "<//retroarch>/retroarch/cheats" same as above


system_directory = "<retroarch>/bios"

system_directory = "default" I named it bios instead of system, but this is the line on the OP. I agree it should be on the XDG_CONFIG_HOME (along with the others here obviously).


savefile_directory = "<retroarch>/saves"

savefile_directory = "default" 

< savestate_directory = "<retroarch>/savestates"

< savestate_directory = "default"

These two should be on there too.


Ok, as you can see it's mildily easy to find 'problems' with a new config file on linux. First there is the minor bug of the duplicate '/'.
Then much more serious several critical settings have just 'default', which i suspect means the file gets created... on the current dir. Most of these (all?) should just use directories on the config home (the dir where the main retroarch.cfg is).

**Comment by j-selby** [comment]
@i30817 please leave issues with other applications elsewhere. Thanks!

**Comment by j-selby** [comment]
@fr500 different option - Citra currently stores all of its files, system and saves in the RetroArch saves directory (or equiv. upstream place, optionally). They are asking for Citra to store just its system files elsewhere.

**Comment by j-selby** [comment]
I'm not going to change the internal directory layout @5schatten - I'm not going to break upstream compatibility for saves for something like that.

**Comment by 5schatten** [comment]
Well didn't you already changed it? I'm pretty sure all system dump files have been located in `~/.local/share/citra-emu` in the past as this diagram points out https://citra-emu.org/wiki/user-directory/

Anyway it's IMHO a source of error since users have to dump systems files to several directories.

**Comment by Papermanzero** [comment]
Yes, it is inconsistent. This was the reason why I opened the issue

**Comment by lilo-san** [comment]
This is annoying me too. Just a suggestion @j-selby. Instead of breaking the internal directory layout in favour of retroarch layout. Could it be proposed to make this directory layout configurable by a config file loaded at startup? So this configurability feature is first implemented by upstream and then merged here. That should make everybody happy.

**Comment by klepp0906** [comment]
just found this myself.  i'd prefer some consistency as well.  took me a minute to find out where sysdata was supposed to go, evidently under %appdata%\citra in windows as placing it with everything else that goes in saves\citra\citra in the RA directory results in it still creating the folder in appdata.  

as noted, ppsspp handles the separation well enough.  

I dont use citra standalone and use retroarch entirely portable so having to deal with files in hidden folders on my system drive is obscure to say the least.
